### PR TITLE
PLAT-1129 | Style voucher

### PR DESCRIPTION
**Pull request by bognix**

### CODE REVIEW

File: app/Traits/CheckoutTrait.php [hunk 1]
```diff
@@ -37,14 +37,12 @@ private function getProduct(Request $request): ?Product
...
+		$coupon = $user ? $user->coupons->last() : null;
 
 		if (session()->has('coupon')) {
 			$sessionCoupon = session()->get('coupon');
 
-			if (!$sessionCoupon instanceof Coupon) {
-				$coupon = null;
-			} else {
+			if ($sessionCoupon instanceof Coupon) {
 				$coupon = session()->get('coupon')->fresh();
```
Comment (rogatty) [inline app/Traits/CheckoutTrait.php]:
```suggestion
				$coupon = $sessionCoupon->fresh();
```

File: resources/views/payment/voucher.blade.php [hunk 2]
```diff
@@ -1,41 +1,40 @@
...
-			@endif
-			<div class="voucher-code">
-				<form action="{{ route('payment-voucher-post') }}" method="post">
+	<div class="t-checkout">
+		@include('payment.cart')
+		<div class="t-checkout__content">
+			<div class="o-voucher o-column">
+				<header class="o-column__row">
+					<h2 class="o-voucher__headline">Mask kod? Super 🎉</h2>
+					<h2 class="o-voucher__headline">Wpisz go poniżej, aby wykorzystać zniżkę.</h2>
```
Comment (rogatty) [inline resources/views/payment/voucher.blade.php]: This looks semantically incorrect. I'd make it two `p`s inside `h2` or without `h2` at all.

File: app/Http/Controllers/Payment/VoucherController.php [hunk 3]
```diff
@@ -21,4 +21,5 @@ public function handle(UseCoupon $request)
 
 		return redirect()->route('payment-account');
 	}
+
```
Comment (bognix) [inline app/Http/Controllers/Payment/VoucherController.php]: remove

File: config/laravel-form-builder.php [hunk 4]
```diff
@@ -49,6 +49,6 @@
 	'default_namespace' => '',
 
 	'custom_fields' => [
-//        'datetime' => App\Forms\Fields\Datetime::class
+		//'datetime' => App\Forms\Fields\Datetime::class
```
Comment (bognix) [inline config/laravel-form-builder.php]: revert

File: resources/assets/sass/styleguide/organisms/_column.scss [hunk 5]
```diff
@@ -0,0 +1,12 @@
+.o-column {
```
Comment (bognix) [inline resources/assets/sass/styleguide/organisms/_column.scss]: not sure what it is. I decided to use organism but it sounds more like a template, but then we agreed that templates are page level things.

File: resources/views/payment/account-buy-album.blade.php [hunk 6]
```diff
@@ -2,21 +2,21 @@
 
 @section('content')
 	<div class="t-checkout">
-		<div class="payment-content t-account">
-			<div class="t-account__row -catalinaBlue">@lang('payment.account-name-heading')</div>
-			<div class="t-account__row">
+		<div class="payment-content t-account o-column">
```
Comment (bognix) [inline resources/views/payment/account-buy-album.blade.php]: check if `payment-content` class could be removed.

File: public/images/voucher_page_hero.svg [hunk 7]
```diff
@@ -0,0 +1 @@
...8,18,0,0,1,0,3.6A1.55,1.55,0,0,1,120,30.61c-1.08.1-1.29-.6-1.44-1.36a133.58,133.58,0,0,0-4.3-14.56c-.4-1.25-1-2.45-1.48-3.66a4,4,0,0,0-1.83-2.14c0-.46.24-.36.5-.26a4.67,4.67,0,0,1,1.79,2.18c1.39,2.93,2.14,6.08,3.29,9.09a76.67,76.67,0,0,1,2.36,8.64c.07.27.14.54.22.81s.23.88.84.83a1,1,0,0,0,.94-.93,12.6,12.6,0,0,0,0-3.47c-.37-3.85-.81-7.69-.91-11.57a3.79,3.79,0,0,1,1.13-2.79,1.49,1.49,0,0,1,2.61.32c.49.95.9,1.94,1.34,2.91C124.9,14.87,125.11,15.22,124.81,15.4Z"/><path class="cls-36" d="M229.73,81.22a28,28,0,0,0-.21,3.88.79.79,0,0,1-.39,1.12,3.65,3.65,0,0,1-.16-2.5c0-1.88,0-3.76.07-5.63a13.48,13.48,0,0,0-1-5.48,4.73,4.73,0,0,0-.65-1.13c-.56-.73-1.13-.66-1.51.17a6,6,0,0,0-.27,1.88c-.54,5.51-.45,11.07-1.18,16.56-.26,2-1.08,2.41-2.58,1.57-.05-.22,0-.35.25-.26,1,.24,1.51-.2,1.64-1.17a23.25,23.25,0,0,0,.42-3.08c0-4.11.41-8.2.76-12.29a18.3,18.3,0,0,1,.17-2.86,1.86,1.86,0,0,1,1.24-1.47,1.46,1.46,0,0,1,1.74.75,11.23,11.23,0,0,1,1.44,4.25.92.92,0,0,1,.29.75C229.56,77.93,229.92,79.57,229.73,81.22Z"/><path class="cls-37" d="M291.82,91.38c2-1.83,4.15-3.47,6.32-5.1A24.43,24.43,0,0,0,303,81.22c2-2.63,4.12-5.08,6.17-7.64a6.58,6.58,0,0,0,.61-.88c.62-1.18.05-2.1-1.27-2a6.85,6.85,0,0,0-3.73,1.11c-.31.2-
```
Comment (rogatty) [inline public/images/voucher_page_hero.svg]: How about cherry-picking https://github.com/bethinkpl/wnl-platform/commit/56324112bf276cf492db12bb88a42cf5843cd1a5 and moving it to `resources/svg`? Also, it's huge, maybe a PNG would be smaller?

Comment (bognix) [reply public/images/voucher_page_hero.svg]: I'll ask Luk for PNG so we can compare. I won't cherry-pick commit from onboarding work, this work is not yet tested and I don't want add an extra level of complexity to this branch.

File: resources/views/payment/voucher.blade.php [hunk 8]
```diff
@@ -1,41 +1,40 @@
...
 						</button>
-					</p>
-					<p class="voucher-skip has-text-centered">
-						<a href="{{ route('payment-account') }}">
-							@lang('payment.voucher-skip')
-						</a>
-					</p>
+					</div>
+					<a href="{{ route('payment-account') }}" class="a-linkInText -stormGray -textMinus1 -textCenter">
+						@lang('payment.voucher-skip')
```
Comment (kvas-damian) [inline resources/views/payment/voucher.blade.php]: This message was designed for old checkout flow. Please update it before merging.

File: public/images/voucher_page_hero.svg (hunk 7 above)
Comment (bognix) [reply public/images/voucher_page_hero.svg]: The PNG is bigger than SVG.

**Pull request merged**